File: Examples/UI/WinForm/VB.NET/TreeList/NFileExplorerUC.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Windows.Forms

Imports Nevron.UI
Imports Nevron.UI.WinForm
Imports Nevron.UI.WinForm.Controls
Imports Nevron.Examples.Framework.WinForm

Namespace Nevron.Examples.UI.WinForm
	Public Partial Class NFileExplorerUC
		Inherits NExampleUserControl
		#Region "Constructor"

		Public Sub New()
			InitializeComponent()
		End Sub
		Public Sub New(ByVal f As MainForm)
			MyBase.New(f)
			InitializeComponent()
		End Sub

		#End Region

		#Region "Overrides"

		Public Overrides Sub Initialize()
			MyBase.Initialize()

			Dock = DockStyle.Fill

			m_FileExplorer = New NFileExplorer()
			m_FileExplorer.AutoSizeColumns = False
			m_FileExplorer.Dock = DockStyle.Fill

			m_FileExplorer.Populate()

			m_FileExplorer.Parent = Me
		End Sub

		#End Region

		#Region "Fields"

		Friend m_FileExplorer As NFileExplorer

		#End Region
	End Class
End Namespace

File: Examples/UI/WinForm/CSharp/TreeList/NFileExplorerUC.cs
using System;
using System.Collections;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Windows.Forms;

using Nevron.UI;
using Nevron.UI.WinForm;
using Nevron.UI.WinForm.Controls;
using Nevron.Examples.Framework.WinForm;

namespace Nevron.Examples.UI.WinForm
{
    public partial class NFileExplorerUC : NExampleUserControl
    {
        #region Constructor

        public NFileExplorerUC()
        {
            InitializeComponent();
        }
        public NFileExplorerUC(MainForm f) : base(f)
        {
            InitializeComponent();
        }

        #endregion

        #region Overrides

        public override void Initialize()
        {
            base.Initialize();

            Dock = DockStyle.Fill;

            m_FileExplorer = new NFileExplorer();
            m_FileExplorer.AutoSizeColumns = false;
            m_FileExplorer.Dock = DockStyle.Fill;

            m_FileExplorer.Populate();

            m_FileExplorer.Parent = this;
        }

        #endregion

        #region Fields

        internal NFileExplorer m_FileExplorer;

        #endregion
    }
}
